File: 프로그래머스/2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes: [[String]]) -> Int {
    var clothesDict: [String: Int] = [:]
    
    for item in clothes {
        let type = item[1]
        clothesDict[type, default: 0] += 1
    }
    
    var combinations = 1
    
    for count in clothesDict.values {
        combinations *= (count + 1)
    }
    
    return combinations - 1
}

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let answer = 1
    const typeCount = new Map()
    for(const [name, type] of clothes){
        typeCount.set(type, (typeCount.get(type) || 0) + 1);
    }
    for(const count of typeCount.values()){
        answer *= (count + 1)
    }
    return answer - 1;
}